src/lib.rs

```diff
@@ -65,6 +67,22 @@ pub trait Body {
     fn size_hint(&self) -> SizeHint {
         SizeHint::default()
     }
+
+    /// Returns future that resolves to next data chunk, if any.
+    fn next_data(&mut self) -> NextData<'_, Self>
+    where
+        Self: Unpin,
+    {
+        NextData(self)
+    }
+
+    /// Returns future that resolves to next data chunk, if any.
+    fn next_trailers(&mut self) -> NextTrailers<'_, Self>
```

Comment:
There's only ever 1 trailers returned, so "next" doesn't need to be in the name.

Since it means the end of the body, should it be by-value? I've been going back and forth on this question for a while.

Reply:
I don't personally need trailers so I didn't investigate much into how you use it.
But if it is last part of body, then it might make some sense.
On other hand, you can poll it despite that `poll_trailers` takes reference, so Next here reflect it

Reply:
Yes, `poll_trailers` takes a reference, but for the same reason `Future::poll` does: it might not be ready yet. But once you've received trailers, you won't receive anymore, so the concept of "next" trailers makes less sense.

Reply:
Its fine for it to take a reference, you'll keep getting `None`. So `trailers` and `next` to me make sense.